fmriprep/workflows/anatomical.py

```diff
@@ -420,60 +422,86 @@ def init_anat_template_wf(longitudinal, omp_nthreads, num_t1w, name='anat_templa
+        (t1_template_dimensions, t1_conform, [
+            ('t1w_valid_list', 'in_file'),
+            ('target_zooms', 'target_zooms'),
+            ('target_shape', 'target_shape')]),
+        (t1_template_dimensions, outputnode, [('out_report', 'out_report'),
+                                              ('t1w_valid_list', 't1w_valid_list')]),
+    ])
+
+    if num_t1w == 1:
+        def _get_first(in_list):
+            if isinstance(in_list, (list, tuple)):
+                return in_list[0]
+            return in_list
+
+        outputnode.inputs.template_transforms = [pkgr('fmriprep', 'data/itkIdentityTransform.tfm')]
+
+        workflow.connect([
+            (t1_conform, outputnode, [(('out_file', _get_first), 't1_template')]),
+        ])
+
+        return workflow
+
+    # 1. Template (only if several T1w images)
+    # 1a. Correct for bias field: the bias field is an additive factor
+    #     in log-transformed intensity units. Therefore, it is not a linear
+    #     combination of fields and N4 fails with merged images.
+    # 1b. Align and merge if several T1w images are provided
+    n4_correct = pe.MapNode(
+        N4BiasFieldCorrection(dimension=3, copy_header=True),
+        iterfield='input_image', name='n4_correct',
+        n_procs=1)  # n_procs=1 for reproducibility
     t1_merge = pe.Node(
-        # StructuralReference is fs.RobustTemplate if > 1 volume, copying otherwise
-        StructuralReference(auto_detect_sensitivity=True,
-                            initial_timepoint=1,      # For deterministic behavior
-                            intensity_scaling=True,   # 7-DOF (rigid + intensity)
-                            subsample_threshold=200,
-                            fixed_timepoint=not longitudinal,
-                            no_iteration=not longitudinal,
-                            transform_outputs=True,
-                            ),
+        fs.RobustTemplate(auto_detect_sensitivity=True,
+                          initial_timepoint=1,      # For deterministic behavior
+                          intensity_scaling=True,   # 7-DOF (rigid + intensity)
+                          subsample_threshold=200,
+                          fixed_timepoint=not longitudinal,
+                          no_iteration=not longitudinal,
+                          transform_outputs=True,
+                          ),
         mem_gb=2 * num_t1w - 1,
         name='t1_merge')
 
-    # Reorient template to RAS, if needed (mri_robust_template may set to LIA)
+    # 2. Reorient template to RAS, if needed (mri_robust_template may set to LIA)
     t1_reorient = pe.Node(Reorient(), name='t1_reorient')
 
-    lta_to_fsl = pe.MapNode(fs.utils.LTAConvert(out_fsl=True), iterfield=['in_lta'],
-                            name='lta_to_fsl')
-
     concat_affines = pe.MapNode(
-        ConcatAffines(3, invert=True), iterfield=['mat_AtoB', 'mat_BtoC'],
+        ConcatAffines(1 + min(num_t1w, 2), invert=True),
```

Comment:
I think you can revert `lta_to_fsl` and `concat_affines` entirely, cleaning up the diff a little.

In any event, this no longer needs the `1 + min(...)` construct.